docs/examples/userguide/wrapping_CPlusPlus/rect_ptr.pyx

```diff
@@ -5,7 +5,12 @@ from Rectangle cimport Rectangle
 cdef class PyRectangle:
     cdef Rectangle*c_rect  # hold a pointer to the C++ instance which we're wrapping
 
-    def __cinit__(self, int x0, int y0, int x1, int y1):
+    def __cinit__(self):
+        self.c_rect = new Rectangle()
+
+    def __init__(self, int x0, int y0, int x1, int y1):
```

Comment:
I disagree with this change I think. In typical use it'll create and destroy a pointless rectangle.

I definitely get the point of preferring `__init__` but in cases like this where `__cinit__` is the right choice, I don't see the point in also providing `__init__`

Reply:
The problem is that, what if later on, the programmer wants to inherit from PyRectangle and create a PySquare that only take in x0, x1, and width as parameters. Using \_\_cinit__ would prevent that possibility. Perhaps, instead of creating a new rectangle, I will just change the attributes of the existing rectangle.

Reply:
> inherit from PyRectangle and create a PySquare

That's a bad design because it violates the Liskov Substitution Principle. :-)

Reply:
Rather, I could imagine allowing `*args*`  and `**kwargs` in `__cinit__` to accommodate for subclasses.

Reply:
However, the way it's structured now is a very good example: Safe initialisation in a bare `__cinit__`, proper value setting in `__init__`, controllable by subclasses. Let's keep it that way.